src/widget/woverview.cpp

```diff
@@ -338,6 +338,8 @@ void WOverview::paintEvent(QPaintEvent * /*unused*/) {
             }
 
             painter.drawImage(rect(), m_waveformImageScaled);
+            // desaturate the scaled waveform-image up to the current play-position by overdrawing semi-transparent black
+            painter.fillRect(0, 0, m_iPos, m_waveformImageScaled.height(), m_signalColors.getPlayedOverlayColor());
```

Comment:
This will only work with horizontal waveform. Please handle the vertical waveform case also. You can use https://github.com/ninomp/mixxx/tree/verticalwaveform_skin for testing.